core/trino-spi/src/main/java/io/trino/spi/block/RowBlockBuilder.java

```diff
@@ -250,8 +255,8 @@ public BlockBuilder appendStructure(Block block)
                 block.writePositionTo(i, fieldBlockBuilders[i]);
             }
         }
-
-        closeEntry();
+        entryAdded(false);
+        currentEntryOpened = false;
```

Comment:
you can remove `currentEntryOpened = true;` and `currentEntryOpened = false;` lines. They are noops in this context

Reply:
done